docs/docs/guides/code-snippets.md

```diff
@@ -0,0 +1,346 @@
+For example, here are the opening lines of the 
+[ReadWriteData.swift](https://github.com/mongodb/docs-realm/blob/master/examples/ios/Examples/ReadWriteData.swift) 
+file:
+
+```swift
+// :replace-start: {
+//   "terms": {
+//     "ReadWriteDataExamples_": ""
+//   }
+// }
+import XCTest
+import RealmSwift
+
+// :code-block-start: models
+class ReadWriteDataExamples_DogToy: Object {
+    @Persisted var name = ""
+}
+
+class ReadWriteDataExamples_Dog: Object {
+    @Persisted var name = ""
+    @Persisted var age = 0
+    @Persisted var color = ""
+    @Persisted var currentCity = ""
+
+    // To-one relationship
+    @Persisted var favoriteToy: ReadWriteDataExamples_DogToy?
+}
+
+class ReadWriteDataExamples_Person: Object {
+    @Persisted(primaryKey: true) var id = 0
+    @Persisted var name = ""
+
+    // To-many relationship - a person can have many dogs
+    @Persisted var dogs: List<ReadWriteDataExamples_Dog>
+
+    // Inverse relationship - a person can be a member of many clubs
+    @Persisted(originProperty: "members") var clubs: LinkingObjects<ReadWriteDataExamples_DogClub>
+}
+
+class ReadWriteDataExamples_DogClub: Object {
+    @Persisted var name = ""
+    @Persisted var members: List<ReadWriteDataExamples_Person>
+}
+// :code-block-end:
+// Many more lines of code examples, until eventually, we end the replace
+// :replace-end:
+```
+
+As you can see, the model names such as `ReadWriteDataExamples_DogToy` are
+very awkward. The `ReadWriteDataExamples` is present to avoid namespace
+collisions with `Dog` or `Person` models in other test files. But this
+awkward name isn't something we want to show documentation viewers.
+
+Fortunately, replace lets us swap any instance of the term we specify with
+some alternative. In this example, we replace `ReadWriteDataExamples`
+with an empty string. 
+
+```swift
+// :replace-start: {
+//   "terms": {
+//     "ReadWriteDataExamples_": ""
+//   }
+// }
+```
+
+The [output file for the code block above](https://github.com/mongodb/docs-realm/blob/master/source/examples/generated/code/start/ReadWriteData.codeblock.models.swift) 
+looks like:
+
+```swift
+class DogToy: Object {
+    @Persisted var name = ""
+}
+
+class Dog: Object {
+    @Persisted var name = ""
+    @Persisted var age = 0
+    @Persisted var color = ""
+    @Persisted var currentCity = ""
+
+    // To-one relationship
+    @Persisted var favoriteToy: DogToy?
+}
+
+class Person: Object {
+    @Persisted(primaryKey: true) var id = 0
+    @Persisted var name = ""
+
+    // To-many relationship - a person can have many dogs
+    @Persisted var dogs: List<Dog>
+
+    // Inverse relationship - a person can be a member of many clubs
+    @Persisted(originProperty: "members") var clubs: LinkingObjects<DogClub>
+}
+
+class DogClub: Object {
+    @Persisted var name = ""
+    @Persisted var members: List<Person>
+}
+```
+
+The long, awkward name has been replaced with nothing.
```

Comment:
Might be worth advising people to be careful to use VerySpecificNamesAndCharacters_ here so that they're sure they don't accidentally delete some common thing.